### ISSUE TITLE
Fetch/Push/Pull button hangs on git operation exception.

### ISSUE THREAD

**Issue by deNoor** [desktop/desktop#15257]
### The problem

Sometimes the main action button hangs on current network operation (fetch, push, pull).
![fetch-button-hanged](https://user-images.githubusercontent.com/25015943/189205513-97575078-083d-480d-ab6b-e2dcfae5adcb.png)
Console window reports errors (also added as text below):
![console-error](https://user-images.githubusercontent.com/25015943/189205532-1c73eadb-2f60-4394-94f8-fec189bdaa9a.png)

To unstuck I need to relaunch Github Desktop.
Git server is company internal on-premise Gitlab instance.

It would be better for the app to show me the exception window, unlock the button and allow me to retry action.

### Release version

3.0.7

### Operating system

Windows 10

### Log files

2022-09-08T18:36:09.071Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 1.815s)
2022-09-08T18:36:11.074Z - info: [ui] Executing updateRemoteHEAD: git -c credential.helper= remote set-head -a origin (took 2.003s)
2022-09-08T18:52:00.742Z - error: [ui] Could not load old contents for syntax highlighting
Error: Git returned an unexpected exit code '128' which should be handled by the caller (getPartialBlobContents).'
    at ChildProcess.<anonymous> (C:\Users\\[the-user]\AppData\Local\GitHubDesktop\app-3.0.7\webpack:\\[name]\app\src\lib\git\spawn.ts:136:11)
    at ChildProcess.emit (node:events:390:28)
    at maybeClose (node:internal/child_process:1064:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
2022-09-08T18:52:00.745Z - error: [ui] Could not load new contents for syntax highlighting
Error: Git returned an unexpected exit code '128' which should be handled by the caller (getPartialBlobContents).'
    at ChildProcess.<anonymous> (C:\Users\\[the-user]\AppData\Local\GitHubDesktop\app-3.0.7\webpack:\\[name]\app\src\lib\git\spawn.ts:136:11)
    at ChildProcess.emit (node:events:390:28)
    at maybeClose (node:internal/child_process:1064:16)
    at Socket.<anonymous> (node:internal/child_process:450:11)
    at Socket.emit (node:events:390:28)
    at Pipe.<anonymous> (node:net:687:12)

### Additional context

`git version 2.37.3.windows.1`
`Microsoft Windows [Version 10.0.19044.1889]`
Server: `gitlab v15.2.2`

**Comment by steveward** [comment]
Thanks for the issue @deNoor. We had a similar report in https://github.com/desktop/desktop/issues/12014. 

Is this only happening in GitLab repositories? Do you have any GitHub repositories in GitHub Desktop as well?

**Comment by deNoor** [comment]
On this corporate PC I have only internal Gitlab repositories added by direct links. Not logged in with GitHub credentials at all. I access all of the repos with the same SSH key. Email is the same (global) for all repos.
Sometimes our Gitlab instance might not respond or connection to corporate VPN might be temporary interrupted.

I've also found that my permissions to one of the added repos where revoked on server. Fetching it directly caused explicit errors, but background fetch might cause hidden fails. I've removed this repo.
